TuneUp/TuneUpWindowViewModel.cs

```diff
@@ -356,35 +339,69 @@ private void CurrentWorkspaceModel_NodeRemoved(NodeModel node)
 
         private void OnCurrentWorkspaceChanged(IWorkspaceModel workspace)
         {
-            profilingEnabled = false;
+            // Profiling needs to be enabled per workspace so mark it false after switching
+            isProfilingEnabled = false;
             CurrentWorkspace = workspace as HomeWorkspaceModel;
-            ResetProfiledNodes();
         }
 
         private void OnCurrentWorkspaceCleared(IWorkspaceModel workspace)
         {
-            profilingEnabled = false;
+            // Profiling needs to be enabled per workspace so mark it false after closing
+            isProfilingEnabled = false;
             CurrentWorkspace = viewLoadedParams.CurrentWorkspaceModel as HomeWorkspaceModel;
-            ResetProfiledNodes();
         }
 
         #endregion
 
-        #region Dispose
+        #region Dispose or setup
 
-        public void Dispose()
+        /// <summary>
+        /// When switching workspaces or closing TuneUp extension,
+        /// unset the legacy workspace for profiling
+        /// </summary>
+        /// <param name="workspace"></param>
+        private void UnsetLegacyWorkspaceforProfiling(HomeWorkspaceModel workspace)
         {
-            foreach (var node in CurrentWorkspace.Nodes)
+            workspace.NodeAdded -= CurrentWorkspaceModel_NodeAdded;
+            workspace.NodeRemoved -= CurrentWorkspaceModel_NodeRemoved;
+            workspace.EvaluationStarted -= CurrentWorkspaceModel_EvaluationStarted;
+            workspace.EvaluationCompleted -= CurrentWorkspaceModel_EvaluationCompleted;
+
+            foreach (var node in workspace.Nodes)
             {
                 node.NodeExecutionBegin -= OnNodeExecutionBegin;
                 node.NodeExecutionEnd -= OnNodeExecutionEnd;
             }
+            executedNodesNum = 0;
+        }
 
-            CurrentWorkspace.NodeAdded -= CurrentWorkspaceModel_NodeAdded;
-            CurrentWorkspace.NodeRemoved -= CurrentWorkspaceModel_NodeRemoved;
-            CurrentWorkspace.EvaluationStarted -= CurrentWorkspaceModel_EvaluationStarted;
-            CurrentWorkspace.EvaluationCompleted -= CurrentWorkspaceModel_EvaluationCompleted;
+        /// <summary>
+        /// When switching workspaces or closing TuneUp extension,
+        /// setup the current workspace for profiling
```

Comment:
this comment says current, but this method works for any workspace as its a parameter.